Mage/src/main/java/mage/abilities/ActivatedAbility.java

```diff
@@ -17,31 +19,46 @@ public interface ActivatedAbility extends Ability {
     final class ActivationStatus {
 
         private final boolean canActivate;
-        private final ApprovingObject approvingObject;
+        private final Set<ApprovingObject> approvingObjects;
 
-        public ActivationStatus(boolean canActivate, ApprovingObject approvingObject) {
+        private ActivationStatus(boolean canActivate, Set<ApprovingObject> approvingObjects) {
             this.canActivate = canActivate;
-            this.approvingObject = approvingObject;
+            this.approvingObjects = approvingObjects;
         }
 
         public boolean canActivate() {
             return canActivate;
         }
 
-        public ApprovingObject getApprovingObject() {
-            return approvingObject;
+        public Set<ApprovingObject> getApprovingObjects() {
+            return approvingObjects;
         }
 
         public static ActivationStatus getFalse() {
-            return new ActivationStatus(false, null);
+            return new ActivationStatus(false, new HashSet<>());
         }
 
         /**
          * @param approvingObjectAbility ability that allows to activate/use current ability
          */
         public static ActivationStatus getTrue(Ability approvingObjectAbility, Game game) {
-            ApprovingObject approvingObject = approvingObjectAbility == null ? null : new ApprovingObject(approvingObjectAbility, game);
-            return new ActivationStatus(true, approvingObject);
+            Set<ApprovingObject> approvingObjects = new HashSet<>();
```

Comment:
That code part looks bad:

1. Don't modify objects on read methods (except lazy init, but I don't see any needs in `null` data here at all -- just init field with a list).
2. No data duplication protections aka multiple calls of `getTrue` method. Fill an approving objects list and reading of the status must be diff methods, not in one.

Reply:
BTW, if you used a set to protect from a duplication (`Set<ApprovingObject>`) then it will not work until override `equals` and `hashCode` (`ApprovingObject` don't have it).

Reply:
3. Typo here. You are introduce and modify local variable `Set<ApprovingObject> approvingObjects`, but due code logic you must modify ability's `this.approvingObjects` here. Of course, if I've got the logic right (ability contains ApprovingObject's list and collect/keep it from a diff calls of `ActivationStatus.getTrue` on activation checks). Same mess with other `getTrue` methods. Must use diff names for class fields and local variables.
![shot_230906_081807](https://github.com/magefree/mage/assets/8344157/8e63f083-3c1d-4f95-a774-08423656d0c2)

Reply:
Should be cleaner. Although `withoutApprovingObject` is weird and probably need a followup investigation. I did preserve the old behavior of 'there is no approving object, but the ActivationStatus.canActivate returns true.'